src/main/java/chacha/Storage.java

```diff
@@ -0,0 +1,102 @@
+package chacha;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.io.BufferedReader;
+import chacha.tasks.Deadline;
+import chacha.tasks.Event;
+import chacha.tasks.Task;
+import chacha.tasks.Todo;
+
+public class Storage {
+    private Path path;
+
+    public Storage(String path) {
+        this.path = Paths.get(path);
+    }
+
+    public void saveToFile(TaskList tasks) throws IOException {
+        BufferedWriter writer = Files.newBufferedWriter(path);
+        for (int i = 0; i < tasks.getSize(); i++) {
+            Task task = tasks.get(i);
+            writer.write(taskToText(task));
+            writer.newLine();
+        }
+        writer.close();
+    }
+
+    public ArrayList<Task> load() throws IOException {
+        Path parent = path.getParent();
+        if (!Files.isDirectory(parent)) {
+            Files.createDirectories(parent);
+        }
+        if (!Files.isRegularFile(path)) {
+            Files.createFile(path);
+        }
+
+        ArrayList<Task> tasks = new ArrayList<>();
+        BufferedReader reader = Files.newBufferedReader(path);
+        String line = reader.readLine();
+        while (line != null) {
+            tasks.add(textToTask(line));
+            line = reader.readLine();
+        }
+        reader.close();
+        return tasks;
+    }
+
+    private String taskToText(Task task) {
+        DateTimeFormatter formatter = DateTimeFormatter.ISO_DATE_TIME;
+        String date;
+        if (task.getDate() != null) {
+            date = task.getDate().format(formatter);
+            date = date.replace("T", " ");
+            date = date.substring(0, 16);
+        } else {
+            date = "no date";
+        }
+        return task.getType() + " , " + task.getStatusIcon() + " , " + task.getDescription() + " , " + date;
+    }
+
+    private Task textToTask(String text) {
+        String[] textArray = text.split(" , "); 
+        String type = textArray[0];
+        String isDone = textArray[1];
+        String description = textArray[2];
+        String date = textArray[3];
+        LocalDateTime dateTime = null;
+        System.out.println(date + "date");
+        if (!date.contains("no date")) {
+            System.out.println("entered this if block");
+            
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+            dateTime = LocalDateTime.parse(date, formatter);
+        }  
+        Task task = new Task();
+        switch (type) {
```

Comment:
The indentation for the switch statement does not follow the recommended standard. 'switch' and 'case' should have the same indentation.
![switch statement](https://user-images.githubusercontent.com/83692822/188309062-6ac321d6-6332-4310-b505-0a3bd6a48adc.png)